roles/openshift_storage_glusterfs/tasks/glusterfs_common.yml

```diff
@@ -108,14 +102,19 @@
       name: "glusterfs-{{ glusterfs_name }}"
       files:
       - "{{ mktemp.stdout }}/glusterfs-storageclass.yml"
-  when:
-  - glusterfs_storageclass or glusterfs_s3_deploy
```

Comment:
You still need the StorageClass to be created before you can deploy gluster-s3. You can't assume glusterfs_storageclass will always be true.

Reply:
@jarrpa  
> You still need the StorageClass to be created before you can deploy gluster-s3.

As part of S3 PVC template, we specify a storage class named glusterfs-${CLUSTER_NAME}-s3. 

Ref: https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_storage_glusterfs/files/gluster-s3-pvcs-template.yml#L21
and 
(https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_storage_glusterfs/files/gluster-s3-pvcs-template.yml#L36)

This is created as part of https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_storage_glusterfs/tasks/gluster_s3_deploy.yml#L34

>You can't assume glusterfs_storageclass will always be true.

So, whether glusterfs_storageclass true/false, we still go ahead create a different storageclass and use the same.

Reply:
....right! I forgot about that. Looks good, then. :)